webapp/sources/rudder/rudder-web/src/test/scala/bootstrap/liftweb/checks/migration/TestMigrateCustomProperties.scala

```diff
@@ -0,0 +1,114 @@
+ *
+ * Rudder is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+ *
+ *************************************************************************************
+ */
+
+package bootstrap.liftweb.checks.migration
+
+import com.normation.errors
+import com.normation.inventory.domain.NodeId
+import com.normation.rudder.facts.nodes.CoreNodeFactRepository
+import com.normation.rudder.facts.nodes.MockLdapFactStorage
+import com.normation.rudder.facts.nodes.NoopGetNodesBySoftwareName
+import com.normation.rudder.facts.nodes.QueryContext
+import com.normation.rudder.facts.nodes.SelectFacts
+import com.normation.rudder.tenants.DefaultTenantService
+
+import com.normation.zio.UnsafeRun
+import org.junit.runner.RunWith
+import org.specs2.mutable.*
+import org.specs2.runner.JUnitRunner
+
+import zio.*
+
+@RunWith(classOf[JUnitRunner])
+class TestMigrateCustomProperties extends Specification {
+
+  val mockLdapFactStorage = new MockLdapFactStorage()
+
+  val factRepo: CoreNodeFactRepository = {
+    (for {
+      tenantService <- DefaultTenantService.make(Nil)
+      repo          <- CoreNodeFactRepository.make(mockLdapFactStorage.nodeFactStorage, NoopGetNodesBySoftwareName, tenantService, Chunk())
+    } yield repo).runNow
+  }
+
+  val migration = new CheckMigrateCustomProperties(mockLdapFactStorage.ldap, factRepo, mockLdapFactStorage.acceptedDIT)
+
+  org.slf4j.LoggerFactory
+    .getLogger("nodes")
+    .asInstanceOf[ch.qos.logback.classic.Logger]
+    .setLevel(ch.qos.logback.classic.Level.TRACE)
+
+  sequential
+
+  "Migrating custom" should {
+    /*
+     * Node1 contains:
+     * - custom properties 'datacenter' and 'from_inv'
+     * - node property 'foo'
+     */
+
+    "find some properties to migrate" in {
+      (migration.findNodeNeedingMigration().runNow must containTheSameElementsAs(List(NodeId("node1")))) and
+      (getNode1PropNamesAsSeenByLdap() must containTheSameElementsAs(List("foo", "datacenter", "from_inv"))) and
+      (getNode1PropNamesAsSeenByRepo() must containTheSameElementsAs(List("foo", "datacenter", "from_inv")))
+    }
+
+    "migrate all nodes" in {
+      migration.migrateAll().either.runNow must beRight()
+    }
+
+    "not find any nodes to migrate after migration" in {
+      (migration.findNodeNeedingMigration().runNow must beEmpty) and
+      (getNode1PropNamesAsSeenByLdap() must containTheSameElementsAs(List("foo", "datacenter", "from_inv"))) and
+      (getNode1PropNamesAsSeenByRepo() must containTheSameElementsAs(List("foo", "datacenter", "from_inv")))
+    }
+  }
+
+  def getNode1PropNamesAsSeenByLdap(): Seq[String] = {
+    (for {
+      _ <- errors.effectUioUnit(println(s"****** ldap ********"))
```

Comment:
I don't see the meaning of the println, for debugging purposes even is it really providing information because I don't see nothing else than `**** ldap ****` and `**** repos ****` printed out ?

Reply:
debugging remaining :)